### PR TITLE
Add stake pool getter to REST API

### PR DESCRIPTION
**Pull request by CodeSandwich**
Fixes https://github.com/input-output-hk/jormungandr/issues/546

BLOCKED BY https://github.com/input-output-hk/jormungandr/pull/657 AND https://github.com/input-output-hk/chain-libs/pull/71

### CODE REVIEW

File: jormungandr/src/main.rs [hunk 1]
```diff
@@ -170,9 +170,17 @@ fn start_services(bootstrapped_node: BootstrappedNode) -> Result<(), start_up::E
             transaction_box: fragment_msgbox,
             block_box: block_msgbox,
         };
+        let block0 = bootstrapped_node
+            .blockchain
+            .lock_read()
+            .tip
+            .ledger()
```
Comment (vincenthz) [inline jormungandr/src/main.rs]: I realize it was not made clear in the original issue, but I think this is was not the intended source of information. I think we want to list the stake pool at the moment it has been snapshotted for the current leadership, not the current stake pool.

that being said, we can fix it later and accept this as a putting most of the REST call in place. Or we can also offer both the current tip and the snapshot (possibly with a different rest endpoint)

File: jormungandr/src/rest/v0/handlers.rs [hunk 2]
```diff
@@ -255,3 +255,16 @@ pub fn get_shutdown(context: State<Context>) -> Result<impl Responder, Error> {
         .stop();
     Ok(HttpResponse::Ok().finish())
 }
+
+pub fn get_stake_pools(context: State<Context>) -> Result<impl Responder, Error> {
+    let blockchain = context.blockchain.lock_read();
+    let stake_pool_ids = blockchain
+        .tip
+        .ledger()
```
Comment (vincenthz) [inline jormungandr/src/rest/v0/handlers.rs]: I realize it was not made clear in the original issue, but I think this is was not the intended source of information. I think we want to list the stake pool at the moment it has been snapshotted for the current leadership, not the current stake pools defined in the tip ledger.

that being said, we can fix it later and accept this as a putting most of the REST call in place. Or we can also offer both the current tip and the snapshot (possibly with a different rest endpoint)

**Comment by vincenthz** [comment]
is this still a WIP ?

**Comment by CodeSandwich** [comment]
No, it's blocked, see description

**Comment by vincenthz** [comment]
it has been merged

File: jormungandr/src/rest/v0/handlers.rs (hunk 2 above)
Comment (CodeSandwich) [reply jormungandr/src/rest/v0/handlers.rs]: Are both versions going to be usable?

**Comment by CodeSandwich** [comment]
Only the first one ;)

**Comment by CodeSandwich** [comment]
Thanks, let me rebase and clean up

**Pull request merged**